eng/illink.targets

```diff
@@ -105,8 +103,42 @@
     </ItemGroup>
   </Target>
 
+
+  <PropertyGroup>
+    <ILLinkResourcesSubstitutionIntermediatePath>$(IntermediateOutputPath)ILLink.Resources.Substitutions.xml</ILLinkResourcesSubstitutionIntermediatePath>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(StringResourcesPath)' != ''">
+    <ILLinkSubstitutionsXmls Include="$(ILLinkResourcesSubstitutionIntermediatePath)" />
+  </ItemGroup>
+  
+  <!-- If a library uses string resources, the following target generates a substitution xml that will be embedded on the
+  library so that if a consumer wants to run the linker they can specify a feature switch to strip out all resources
+  from the assembly. -->
+  <Target Name="GenerateResourcesSubstitutionFile"  
+          Condition="'$(StringResourcesPath)' != ''"
+          Inputs="'$(MSBuildProjectFullPath)'"
+          Outputs="'$(ILLinkResourcesSubstitutionIntermediatePath)'">
+
+    <PropertyGroup>
+      <ILLinkResourcesSubstitutionTemplate>$(MSBuildThisFileDirectory)ILLink.Resources.Substitutions.template</ILLinkResourcesSubstitutionTemplate>
+    </PropertyGroup>
+
+    <WriteLinesToFile File="$(ILLinkResourcesSubstitutionIntermediatePath)"
+                      Lines="$([System.IO.File]::ReadAllText('$(ILLinkResourcesSubstitutionTemplate)')
```

Comment:
>ILLinkResourcesSubstitutionTemplate [](start = 64, length = 35)

does this file guaranteed to exist all the time when this target get called?

Reply:
yes, the template is under the /eng folder so will always be there.